### DISCUSSION TITLE
查询条件加上order by后变的很慢

### DISCUSSION THREAD

**Discussion by liuwangfa0512** [apache/doris#9425] [A - General / Q&A]
使用版本1.0.0 
1台fe 8和16G 3台be 16核64G 
数据10亿左右

不使用order 和使用order 差距很大，
建表时使用UNIQUE KEY(`trade_time`,`ts_code`) ，以及使用trade_time 按月动态分区

在未创建bitmap索引的时候排序为3.5s 创建之后如下 有所提高 但是还是慢CREATE INDEX IF NOT EXISTS idx_tick_trade_time ON tick (trade_time) USING BITMAP COMMENT 'trade_time';

mysql> 
mysql> 
mysql> select * from tick where  trade_time > '2022-03-29 14:52:06' and ts_code='000006.SZ'   limit 10000;
10000 rows in set (0.32 sec)

select * from tick where  trade_time > '2022-03-29 14:52:06' and ts_code='000006.SZ' order by trade_time desc  limit 10000;
10000 rows in set (1.61 sec)

select * from tick where  trade_time > '2022-03-29 14:52:06' and ts_code='000006.SZ' order by trade_time  limit 10000;
10000 rows in set (1.32 sec)

**Comment by dataalive** [comment]
这个查询时是否开启了向量化？

**Comment by liuwangfa0512** [comment]
没有开启向量化 或者可以给一下命令我确认一下是否用了


发自我的iPhone

**Reply by hf200012** [reply]
set enable_vectorized_engine = true;
set batch_size = 4096;

**Comment by liuwangfa0512** [comment]
谢谢解答
还有一个问题
创建表的时候指明decimal(14,5)了保留小数位数，但是实际没有补0操作
CREATE TABLE `stock_indicator` (
   `ts_code` varchar(30) NOT NULL COMMENT "股票代码",
   `trade_date` varchar(30) NOT NULL COMMENT "交易日期",
   `trade_date_doris` date NOT NULL COMMENT "日期",
   `close`  decimal(14,5) NULL COMMENT "收盘价",
   `open`  decimal(14,5)  NULL COMMENT "开盘价",
   `high`  decimal(14,5)  NULL COMMENT "最高价",
   `low`  decimal(14,5)  NULL COMMENT "最低价",


mysql> desc stock_indicator;
+------------------+---------------+------+-------+---------+---------+
| Field            | Type          | Null | Key   | Default | Extra   |
+------------------+---------------+------+-------+---------+---------+
| ts_code          | VARCHAR(30)   | No   | true  | NULL    |         |
| trade_date       | VARCHAR(30)   | No   | true  | NULL    |         |
| trade_date_doris | DATE          | No   | true  | NULL    |         |
| close            | DECIMAL(14,5) | Yes  | false | NULL    | REPLACE |
| open             | DECIMAL(14,5) | Yes  | false | NULL    | REPLACE |
| high             | DECIMAL(14,5) | Yes  | false | NULL    | REPLACE |
| low              | DECIMAL(14,5) | Yes  | false | NULL    | REPLACE |
| pre_close        | DECIMAL(14,5) | Yes  | false | NULL    | REPLACE |
| change           | DECIMAL(14,5) | Yes  | false | NULL    | REPLACE |
| pct_change       | DECIMAL(14,5) | Yes  | false | NULL    | REPLACE |






ts_code trade_date close open high low pre_close change pct_change vol
0 000001.SZ 20220520 15.02 14.7 15.04 14.67 14.62 0.4 2.736 948597.78 1 000001.SZ 20220519 14.62 14.55 14.7 14.5 14.75 -0.13 -0.8814 590921.72 2 000001.SZ 20220518 14.75 14.85 14.89 14.67 14.71 0.04 0.2719 778578.33 3 000001.SZ 20220517 14.71 14.51 14.72 14.32 14.41 0.3 2.0819 931406.39















在 2022-05-09 10:13:07，"jiafeng.zhang" ***@***.***> 写道：

set enable_vectorized_engine = true;
set batch_size = 4096;

**Reply by liaoxin01** [reply]
因为现在decimal在BE层采用固定（27,9）的精度来实现的，FE没有传递准确的decimal精度信息到BE，导致输出结果时不知道具体的小数位信息，所以这里就没有进行补零。
我们正在对decimal代码进行重构，新版的decimal能够正确的补零。

**Comment by liuwangfa0512** [comment]
谢谢 期待


发自我的iPhone

**Comment by liuwangfa0512** [comment]
还有像这种问题 一般需要优化什么 谢谢
写入DORIS失败，200 {
    "TxnId": 194854,
    "Label": "7557dba9-20a8-4be5-a02d-3061afc9c2cc",
    "TwoPhaseCommit": "false",
    "Status": "Fail",
    "Message": "errCode = 2, detailMessage = Failed to commit txn 194854. Tablet [35907] success replica num 1 is less than quorum replica num 2 while error backends 10002,10004",
    "NumberTotalRows": 212,
    "NumberLoadedRows": 212,
    "NumberFilteredRows": 0,
    "NumberUnselectedRows": 0,
    "LoadBytes": 134749,
    "LoadTimeMs": 197,
    "BeginTxnTimeMs": 1,
    "StreamLoadPutTimeMs": 4,
    "ReadDataTimeMs": 0,
    "WriteDataTimeMs": 185,
    "CommitAndPublishTimeMs": 0
}













--

刘望发
上海网达软件有限公司
地址：上海市浦东新区新金桥路27号13号楼3层
手机：13801863843
***@***.***
网络无处不在，沟通及时到达




在 2022-05-21 17:03:18，"Xin Liao" ***@***.***> 写道：

因为现在decimal在BE层采用固定（27,9）的精度来实现的，FE没有传递准确的decimal精度信息到BE，导致输出结果时不知道具体的小数位信息，所以这里就没有进行补零。
我们正在对decimal代码进行重构，新版的decimal能够正确的补零。

**Reply by LemonLiTree** [reply]
这个写入副本成功的只有1个，两个失败了，需要具体看下失败原因。

**Comment by liuwangfa0512** [comment]
你好
  在fe.conf 中设置了参数 max_conn_per_user = 2000没有生效
并且在 ADMIN SHOW FRONTEND CONFIG 也找不到这个参数，以及设置的时候也报错 是这个参数没有了吗？




mysql> ADMIN SET FRONTEND CONFIG ("max_conn_per_user" = "500");

ERROR 1105 (HY000): errCode = 2, detailMessage = Config 'max_conn_per_user' does not exist







--

刘望发
上海网达软件有限公司
地址：上海市浦东新区新金桥路27号13号楼3层
手机：13801863843
***@***.***
网络无处不在，沟通及时到达




在 2022-05-21 17:03:18，"Xin Liao" ***@***.***> 写道：

因为现在decimal在BE层采用固定（27,9）的精度来实现的，FE没有传递准确的decimal精度信息到BE，导致输出结果时不知道具体的小数位信息，所以这里就没有进行补零。
我们正在对decimal代码进行重构，新版的decimal能够正确的补零。

**Reply by LemonLiTree** [reply]
> 你好 在fe.conf 中设置了参数 max_conn_per_user = 2000没有生效 并且在 ADMIN SHOW FRONTEND CONFIG 也找不到这个参数，以及设置的时候也报错 是这个参数没有了吗？ mysql> ADMIN SET FRONTEND CONFIG ("max_conn_per_user" = "500"); ERROR 1105 (HY000): errCode = 2, detailMessage = Config 'max_conn_per_user' does not exist
> […](#)
> -- 刘望发 上海网达软件有限公司 地址：上海市浦东新区新金桥路27号13号楼3层 手机：13801863843 ***@***.*** 网络无处不在，沟通及时到达 在 2022-05-21 17:03:18，"Xin Liao" ***@***.***> 写道： 因为现在decimal在BE层采用固定（27,9）的精度来实现的，FE没有传递准确的decimal精度信息到BE，导致输出结果时不知道具体的小数位信息，所以这里就没有进行补零。 我们正在对decimal代码进行重构，新版的decimal能够正确的补零。 — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you authored the thread.Message ID: ***@***.***>

这个参数已经失效了
